[New Resource]: Ethernet Port

acch opened the pull request:
Adds ability to create ethernet ports (VLAN / LAG).

Closes #304.

Acceptance tests pass:
```shell
TF_ACC=1 go test ./internal/provider/networking/network_ethernet_port_resource_test.go -v
=== RUN   TestAccNetworkEthernetPortResource
--- PASS: TestAccNetworkEthernetPortResource (16.33s)
PASS
ok      command-line-arguments  16.338s
```

Example Terraform Configurations:
```terraform
// sample LAG with required properties
resource "netapp-ontap_port" "lag" {
  cx_profile_name = ...

  broadcast_domain = {
    ipspace = "Default"
    name    = "bd1"
  }

  node = {
    name = "node1"
  }

  type = "lag"
  lag = {
    distribution_policy = "mac"
    member_ports = [ "e0b", "e0c" ]
    mode = "singlemode"
  }
}
```

```terraform
// sample VLAN with required properties
resource "netapp-ontap_port" "vlan" {
  cx_profile_name = ...

  broadcast_domain = {
    ipspace = "Default"
    name    = "bd2"
  }

  node = {
    name = "node1"
  }

  type = "vlan"
  vlan = {
    base_port = "e0a"
    tag       = 100
  }
}
```

acch committed:
feat: add `ethernet_port` data source

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
fix: rename data source to `port`

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: add `ports` data source

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: use Terraform Set Type instead of Go built-in slice

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: add `port` resource - create method

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: consistent error handling

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: add LAG examples

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: add `port` resource - delete method

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
fix: correct 'port' resource for type LAG

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: add remaining attribute description

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: finalize data models with remaining attributes

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: add acceptance tests for 'port' resource - read, create, delete methods

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: add `port` resource - update method

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: improve create method so that it populates all resource attributes without requiring refresh

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: add `port` resource - import method

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
chore: change cx_profile_name for consistency with other examples

Signed-off-by: Achim Christ <achim.christ@sva.de>

carchi8py reviewed (changes requested):
Looks good as well just missing documentation files.

acch committed:
feat: add `port` documentation

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch commented:
Thanks @carchi8py for your feedback!

I've manually added the documentation pages in the `docs/` directory.

acch commented:
@carchi8py Are we good to merge this PR?

Please let me know what I need to do exactly to have the PRs added to the provider. My understanding is that I've addressed all requested changes... please let me know what's missing.

Many thanks!

suhasbshekar commented:
Can you update CHANGELOG.md file?

Also your branch is behind the integration/main, we added few more checks to the pipeline as it was missing here, since your PR is from a fork of this repo, so could you pull the new changes into your branch and update the PR?

acch commented:
@suhasbshekar Thanks for your feedback.

- I've added an entry to CHANGELOG.md
- I've rebased my branch to integration/main (once again)

Acc tests are passing on my local test system (ONTAP Select running 9.15.1P4):

```shell
$ date && TF_ACC=1 go test ./internal/provider/networking/network_ethernet_port_resource_test.go -v && date
Tue Feb  4 09:36:52 UTC 2025
=== RUN   TestAccNetworkEthernetPortResource
--- PASS: TestAccNetworkEthernetPortResource (15.37s)
PASS
ok      command-line-arguments  15.375s
Tue Feb  4 09:37:08 UTC 2025
```

Please let me know what else is needed.
Many thanks!

acch committed:
chore: update changelog

Signed-off-by: Achim Christ <achim.christ@sva.de>

suhasbshekar commented:
Please resolve the conflicts.

acch commented:
@suhasbshekar I've rebased the PR once again.

Can you please let me know when you'll be able to merge these changes?

suhasbshekar commented:
Conflicts

acch commented:
Resolved
